### ISSUE TITLE
read_graphml function rejecting well formed Graphml files

### ISSUE THREAD

**Issue by bhayat-quantinuum** [boostorg/graph#369]
Here is a MWE of my code:

```
#include <iostream>
#include <fstream>
#include <boost/graph/graphml.hpp>
#include <boost/property_map/dynamic_property_map.hpp>

int main(int, char**){

    std::ifstream graph_file;
    graph_file.open("more_exampleHoles_8.txt", std::ifstream::in);

    Graph g;
    boost::dynamic_properties dp;
    boost::read_graphml(graph_file, g, dp);
}
```

The file `more_exampleHoles_8.txt` is attached below. I receive the following error:

```
terminate called after throwing an instance of 'boost::wrapexcept<boost::property_not_found>'
  what():  Property not found: betti.
Aborted (core dumped)
```

I think this is a bug? If not, any help appreciated.

[more_exampleHoles_8.txt](https://github.com/boostorg/graph/files/14863835/more_exampleHoles_8.txt)

**Comment by jeremy-murphy** [comment]
Do you have something else that reads GraphML to compare against? I mean, how did you verify that it's well formed?

**Comment by andreacassioli** [comment]
If it can help, I have tried with python NetworkX package and the example seems legit, no error. However boost does indeed raises an exception.

**Comment by jeremy-murphy** [comment]
Thanks for the confirmation.

**Comment by Becheler** [comment]
Quick reproducer here: https://godbolt.org/z/MvfboxPzn

GraphML standard states that if a node has no `<data>` element for a key, the key's default value applies (or the property simply doesn't apply to that node): omitting `<data>` for a key is valid.

NetworkX silently accepts any property it encounters and stores it dynamically. If a node has no data for a key, it just doesn't store anything for that node and does not throw.

But BGL requires every property name encountered in `<data>` elements to have a pre-registered property map in `dynamic_properties`. If it finds a `<data key="v_betti">` and `dp` has no map for `"betti"`, then it throws `property_not_found`. 

So in short, the file is valid, but BGL contract requires to pre-register all properties they care about or to use `ignore_other_properties` to opt out.

This was a common complaint so the new documentation added a tip about it in #491  : https://491.graph.prtest3.cppalliance.org/graph/io/graphml.html#_read_graphml

I will close it as completed, but don't hesitate reopening if it did not solve the issue 😄
